tests/test_gls_fitter.py

```diff
@@ -60,6 +60,20 @@ def test_gls_fitter(self):
                     assert np.abs(v - val[0]) <= e, msg
                     assert np.abs(1 - val[1] / e) < 0.1, msg
 
+    def test_noise_design_matrix_index(self):
```

Comment:
Does this test fail when run against the code before the bug fix? I can't really tell what it is testing or whether it actually verifies correctness.